HPXMLtoOpenStudio/resources/waterheater.rb

```diff
@@ -979,16 +962,26 @@ def self.add_hpwh_control_program(model, obj_name_hpwh, amb_temp_sensor, hpwh_bo
 
   def self.get_combi_boiler_and_plant_loop(model, heating_source_id)
     # Search for the right boiler OS object
-    boiler = nil
-    plant_loop = nil
+    boiler_hw = nil
+    plant_loop_hw = nil
     model.getBoilerHotWaters.each do |bhw|
       sys_id = bhw.additionalProperties.getFeatureAsString('HPXML_ID')
-      if sys_id.is_initialized && sys_id.get == heating_source_id
-        boiler = bhw
-        plant_loop = boiler.plantLoop.get
+      next unless sys_id.is_initialized && sys_id.get == heating_source_id
+
+      plant_loop = bhw.plantLoop.get
+      plant_loop_hw = plant_loop.clone(model).to_PlantLoop.get
+      # pump power is zero, fixme: is this expected? what about no heating scenarios when only water heating boiler is running but no power consumption is calculated?
+      plant_loop_hw.supplyComponents.each do |comp|
+        if comp.to_BoilerHotWater.is_initialized
+          boiler_hw = comp.to_BoilerHotWater.get
+        end
+        next unless comp.to_PumpVariableSpeed.is_initialized
+
+        pump_hw = comp.to_PumpVariableSpeed.get
+        pump_hw.setRatedPowerConsumption(0.0)
```

Comment:
Can you explain this too? I don't understand the reason for this code nor your comment above it.

Reply:
This is the assumption we make for water heating boiler that pump power is not set (0 for now). This will underestimate pump energy consumption especially when in summer if only water heating is needed.

Reply:
There shouldn't be any pump energy associated with water heating, the water coming from the mains is pressurized and you just need to open the tap for it to come out. There is pump energy associated with space heating because it's a closed loop system and the pump is used to circulate the hot water to your radiators/baseboards.